### PR TITLE
Add torch version of MQ-CNN and MQ-RNN

### PR DESCRIPTION
**Pull request by lostella**
*Issue #, if available:* Addressing #2059

*Description of changes:* Adding PyTorch version of MQ-CNN and MQ-RNN, with lightning module around it for training purposes. Work in progress, notable features missing:
- [ ] "forking" behaviour for loss computation
- [x] using categorical features 
- [x] provide scale as input to the encoder
- [x] estimator class
- [ ] validating correctness wrt paper/mxnet implementation
- [ ] running experiments to check accuracy compared to mxnet implementation

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Please tag this pr with at least one of these labels to make our release process faster:** BREAKING, new feature, bug fix, other change, dev setup

### CODE REVIEW

**Comment by timoschowski** [comment]
any chance that this is going in before the Sphere course starts on Jan 17? That would be rather awesome!
